mmap-alloc/src/lib.rs

```diff
@@ -81,6 +102,35 @@ use errno::errno;
 /// In order to avoid this scenario, it is often necessary to explicitly flush the instruction
 /// cache before executing newly-written memory. The mechanism to accomplish this differs by
 /// system.
+///
+/// # Alignment
+///
+/// Since memory is allocated by mapping pages directly from the kernel, all allocations are
+/// automatically aligned to the size of a memory page. Since there is no way to request a
+/// higher alignment from the kernel, higher alignments are not supported by `MapAlloc` by default.
+///
+/// However, this crate provides the `large-align` feature (disabled by default) which works around
+/// this limitation to allow for arbitrarily large alignments. When an alignment larger than the page
+/// size is requested, `alignment - page size` extra bytes are allocated. There is guaranteed to be
+/// a properly-aligned chunk of memory of the requested size somewhere within this larger allocation.
+/// That chunk is located and returned. The `large-align` feature also provides an `object_align`
+/// method on `MapAllocBuilder` that configures the alignment of the `UntypedObjectAlloc` implementation.
+///
+/// ## Platform-specific behavior
+///
+/// On Linux and Mac, all extra leading and trailing memory surrounding the located chunk is
+/// deallocated, leaving only the requested object allocated. On Windows, that extra memory is left
+/// allocated, but is left in the uncommitted state. This should ensure that it does not consume any
+/// physical resources, although it will still consume virtual memory space, which may be scarce,
+/// especially on 32-bit platforms.
+///
+/// ## Limitations
+///
+/// Currently, the `large-align` feature does not support large alignments in the `realloc` method,
```

Comment:
Why isn't this supported?

Reply:
It's just a pain in the butt to implement and we don't need it for our use case in elfmalloc.